### PR TITLE
DDF-04410, MapSettings components unit tests with network calls mocking.

### PR DESCRIPTION
**Pull request by dimitry-sherman**
#### What does this PR do?
Addes unit tests to MapSettings components.
#### Who is reviewing it? 
@djblue @andrewkfiedler @Bdthomson 
#### Select relevant component teams: 
@codice/ui 
#### Ask 2 committers to review/merge the PR and tag them here.
@djblue 
@andrewkfiedler

#### How should this be tested?
Run `yarn start:test`
#### What are the relevant tickets?
For Jira:
[](https://codice.atlassian.net/browse/)

For GH Issues:
Fixes: #04410

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [X] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by Bdthomson** [comment]
build noow

**Comment by Bdthomson** [comment]
build now

File: ui/packages/catalog-ui-search/src/main/webapp/test/mock-api/mock-properties.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+const properties = require('../../js/properties')
+const api = require('./index')
+const oldInit = properties.init
+
+const mock = () => {
+  properties.init = function() {
+    const data = api('./internal/config')
+    const uiConfig = api('./internal/platform/config/ui')
+    // use this function to initialize variables that rely on others
```
Comment (Bdthomson) [inline ui/packages/catalog-ui-search/src/main/webapp/test/mock-api/mock-properties.js]: Is this comment still needed?

File: yarn.lock [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (Bdthomson) [inline yarn.lock]: Delete this file.

File: ui/packages/catalog-ui-search/src/main/webapp/test/mock-api/mock-properties.js (hunk 1 above)
Comment (dimitry-sherman) [reply ui/packages/catalog-ui-search/src/main/webapp/test/mock-api/mock-properties.js]: I think it's still useful as to clarify the mandatory nature of this initialization.

**Comment by djblue** [comment]
build now

**Comment by Bdthomson** [comment]
build now

**Comment by dimitry-sherman** [comment]
build now

**Comment by dimitry-sherman** [comment]
build now

**Comment by andrewkfiedler** [comment]
Checked out and verified that running `yarn start:test` in the catalog-ui-search folder worked.

**Pull request merged**